File: leetcode/1894-merge-strings-alternately/solution.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        len1 = len(word1)
        len2 = len(word2)
        maxlen = max(len1,len2)
        merged = []
        i = 0
        for i in range(maxlen):
            if i < len1:
                merged.append(word1[i])
            if i < len2:
                merged.append(word2[i])

        return "".join(merged)

File: leetcode/1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder result = new StringBuilder();

        int l1 = word1.length();
        int l2 = word2.length();
        int min = Math.min(l1,l2);
        int i = 0;

        while(i<min)
        {
            result.append(word1.charAt(i));
            result.append(word2.charAt(i));
            i++;
        }
        if (l1>l2)
        {
            result.append(word1.substring(i));
        }
        else if (l2>l1)
        {
            result.append(word2.substring(i));
        }
        return result.toString();
    }
}
